pkg/datasources/grants.go

```diff
@@ -248,135 +321,193 @@ var grantsSchema = map[string]*schema.Schema{
 
 func Grants() *schema.Resource {
 	return &schema.Resource{
-		Read:   ReadGrants,
-		Schema: grantsSchema,
+		ReadContext: ReadGrants,
+		Schema:      grantsSchema,
 	}
 }
 
-func ReadGrants(d *schema.ResourceData, meta interface{}) error {
+func ReadGrants(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	db := client.GetConn().DB
 
-	var grantDetails []snowflake.GrantDetail
+	var opts *sdk.ShowGrantOptions
 	var err error
-	if v, ok := d.GetOk("grants_on"); ok {
-		grantsOn := v.([]interface{})[0].(map[string]interface{})
-		objectType := grantsOn["object_type"].(string)
-		objectName := grantsOn["object_name"].(string)
-		account := grantsOn["account"].(bool)
+	if grantsOn, ok := d.GetOk("grants_on"); ok {
+		opts, err = buildOptsForGrantsOn(grantsOn.([]interface{})[0].(map[string]interface{}))
```

Comment:
could you replace `interface{}` with `any`?

Reply:
done